server/channels/store/sqlstore/post_store.go

```diff
@@ -2753,7 +2753,7 @@ func (s *SqlPostStore) SearchPostsForUser(paramsList []*model.SearchParams, user
 		if result.NErr != nil {
 			return nil, result.NErr
 		}
-		data := result.Data.(*model.PostList)
+		data := result.Data
 		posts.Extend(data)
```

Comment:
Nit: We can simplify the code here
```suggestion
		posts.Extend(result.Data)
```